phip/clipped_factorization_analysis.py

```diff
@@ -0,0 +1,144 @@
+import time
+
+import pandas
+import numpy
+
+import tensorflow as tf
+from tensorflow.contrib.distributions import percentile
+
+
+def do_clipped_factorization_analysis(
+        counts_df,
+        rank=3,
+        truncate_percentile=99.9,
```

Comment:
How can we determine what the best value for this is?

Reply:
We should be pretty robust to this choice (e.g. 99 vs. 99.9 vs. 99.99). It just needs to be a lower-bound on fraction of non-hits. May be good to try different values though as we analyze the next few data sets